pkgs/by-name/py/pyspread/package.nix

```diff
@@ -6,28 +6,22 @@
 , makeDesktopItem
 }:
 
-let
-  # get rid of rec
+python3.pkgs.buildPythonApplication (finalAttrs: {
   pname = "pyspread";
   version = "2.2.2";
   src = fetchPypi {
-    inherit pname version;
+    pname = "pyspread";
+    inherit (finalAttrs) version;
     hash = "sha256-vbBu/dMXQf14F7qWvyHX5T8/AkjeZhaQt1eQ6Nidpsc=";
```

Comment:
I don't like this pattern, when only changing version it silently makes the FOD none reproducible.

Reply:
Looks like fetchpypi doesn't let you pass in anything that sends up in the source derivation's `name` https://github.com/NixOS/nixpkgs/blob/c36d884e6aafc55f1ce06cfef41f702cb16ece6f/pkgs/build-support/fetchpypi/default.nix#L27 so can't fix it that way

Reply:
In `xrootd.passthru.tests.test-xrdcp`, I pass a `version` to the FOD that consists of `<xrootd version from finalAttrs>-<xrootd drv hash from finalAttrs>` to make it rebuilds everytime `xrootd` gets rebuild.

The `version` trick might also be suitable for source FODs. However, for vendoring FODs, it could be a waste of time and space to rebuild when the vendorHash remains the same.

The ultimate solution might be a mechanism to generate automatic `passthru.tests` for all the FOD passed into `stdenv.mkDerivaiton`. Automatically-generated `passthru.tests` test cases could also be a way to run ShellCheck against build commands (#21166).

Reply:
In the case of `fetchPypi`, the `outPath` changes when `version` changes. The FOD package will be rebuild even when `hash` isn't changed.